src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathMapperTest.java

```diff
@@ -0,0 +1,532 @@
+        final RootNode interfaceRootNode = new RootNode(interfaceRootAst);
+        final DetailAST[] actual = convertToDetailAstArray(getXpathItems(xpath, interfaceRootNode));
+        final DetailAST realInterfaceDefNode = getSiblingByType(interfaceRootAst,
+                TokenTypes.INTERFACE_DEF);
+        final DetailAST[] expected = {realInterfaceDefNode};
+        assertArrayEquals(expected, actual);
+    }
+
+    @Test
+    public void testQueryInterfaceMethodDefNumber() throws Exception {
+        final String xpath = "/INTERFACE_DEF/OBJBLOCK/METHOD_DEF";
+        final DetailAST interfaceRootAst = getRootAst("InputXpathInterface.java");
+        final RootNode interfaceRootNode = new RootNode(interfaceRootAst);
+        final List<Item> nodes = getXpathItems(xpath, interfaceRootNode);
+        assertEquals(4, nodes.size());
+    }
+
+    @Test
+    public void testQueryInterfaceParameterDef() throws Exception {
+        final String xpath = "//PARAMETER_DEF[@text='someVariable']/../..";
+        final DetailAST interfaceRootAst = getRootAst("InputXpathInterface.java");
+        final RootNode interfaceRootNode = new RootNode(interfaceRootAst);
+        final DetailAST[] actual = convertToDetailAstArray(getXpathItems(xpath, interfaceRootNode));
+        final DetailAST realMethodDefNode = getSiblingByType(interfaceRootAst,
+                TokenTypes.INTERFACE_DEF)
+                .findFirstToken(TokenTypes.OBJBLOCK)
+                .findFirstToken(TokenTypes.METHOD_DEF)
+                .getNextSibling();
+        final DetailAST[] expected = {realMethodDefNode};
+        assertArrayEquals(expected, actual);
+    }
+
+    private static DetailAST getRootAst(String fileName) throws Exception {
+        final File file = new File("src/test/resources/com/puppycrawl/tools/"
+                + "checkstyle/xpath/" + fileName);
+        return TestUtils.parseFile(file);
+    }
+
+    private static List<Item> getXpathItems(String xpath, AbstractNode rootNode) throws Exception {
+        final XPathEvaluator xpathEvaluator = new XPathEvaluator();
+        final XPathExpression xpathExpression = xpathEvaluator.createExpression(xpath);
+        final XPathDynamicContext xpathDynamicContext = xpathExpression
+                .createDynamicContext(rootNode);
+        return xpathExpression.evaluate(xpathDynamicContext);
+    }
+
+    private static DetailAST[] convertToDetailAstArray(List<Item> nodes) {
+        final DetailAST[] result = new DetailAST[nodes.size()];
+        for (int i = 0; i < nodes.size(); i++) {
+            final AbstractNode abstractNode = (AbstractNode) nodes.get(i);
+            result[i] = abstractNode.getUnderlyingNode();
+        }
+        return result;
+    }
+
+    private static DetailAST getSiblingByType(DetailAST node, int type) {
+        DetailAST returnValue = null;
+        for (DetailAST ast = node; ast != null; ast = ast.getNextSibling()) {
```

Comment:
> but I need only first level children

Isn't this just `DetailAST.findFirstToken`?
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java#L384